File: ch19p533modelPresentationContextBug/ChainTest/SecondViewController.m


#import "SecondViewController.h"

@interface SecondViewController ()

@end

@implementation SecondViewController
- (IBAction)doDismiss:(id)sender {
    NSLog(@"pvc %@", self.presentingViewController);
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}
- (IBAction)doPresent:(id)sender {
    // work out true presenter
    UIViewController* rvc = self.view.window.rootViewController;
    while (rvc.presentedViewController)
        rvc = rvc.presentedViewController;
    UIViewController* vc = [SecondViewController new];
    [rvc presentViewController:vc animated:YES completion:^{
        // NSLog(@"%@", vc.view.window.rootViewController);
        NSLog(@"pvc %@", vc.presentingViewController);
        NSLog(@"pvc.pvc %@", vc.presentingViewController.presentedViewController);
    }];
}


@end

File: iOS9bookExamples/bk2ch06p271presentedViewControllerAdaptive2/ch19p601presentedViewController/SecondViewController.swift


import UIKit

/* 

Standard architecture for handing info from vc to presented vc...
...and back when presented vc is dismissed

*/

protocol SecondViewControllerDelegate : class {
    func acceptData(data:AnyObject!)
}

class SecondViewController : UIViewController {
    
    var data : AnyObject?
    
    weak var delegate : SecondViewControllerDelegate?
    
    @IBAction func doDismiss(sender:AnyObject?) {
        self.presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        print(self.view.bounds.size.width, terminator: "")
        print("\t", terminator: "")
        print(self.view.bounds.size.height, terminator: "")
        print("\t", terminator: "")

        print(self.view.backgroundColor!)


    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        if self.isBeingDismissed() {
            self.delegate?.acceptData("Even more important data!")
        }
    }
    
    
}
